src/Device.ts

```diff
@@ -85,64 +87,79 @@ export function detectDevice(): BuiltinHandlerName | undefined
 	else if (typeof navigator === 'object' && typeof navigator.userAgent === 'string')
 	{
 		const ua = navigator.userAgent;
-		const browser = Bowser.getParser(ua);
-		const engine = browser.getEngine();
+
+		const uaParser = new UAParser(ua);
+
+		logger.debug(
+			'detectDevice() | browser detected [ua:%s, parsed:%o]',
+			ua, uaParser.getResult()
+		);
+
+		const browser = uaParser.getBrowser();
+		const browserName = browser.name?.toLowerCase() ?? '';
+		const browserVersion = parseInt(browser.major ?? '0');
+		const engine = uaParser.getEngine();
+		const engineName = engine.name?.toLowerCase() ?? '';
+		const os = uaParser.getOS();
+		const osName = os.name?.toLowerCase() ?? '';
+		const osVersion = parseFloat(os.version ?? '0');
 
 		// Chrome, Chromium, and Edge.
-		if (browser.satisfies({ chrome: '>=111', chromium: '>=111', 'microsoft edge': '>=111' }))
+		if ([ 'chrome', 'chromium', 'edge' ].includes(browserName) && browserVersion >= 111)
 		{
 			return 'Chrome111';
 		}
-		else if (browser.satisfies({ chrome: '>=74', chromium: '>=74', 'microsoft edge': '>=88' }))
+		else if (
+			([ 'chrome', 'chromium' ].includes(browserName) && browserVersion >= 74) ||
+			(browserName === 'edge' && browserVersion >= 88)
+		)
 		{
 			return 'Chrome74';
 		}
-		else if (browser.satisfies({ chrome: '>=70', chromium: '>=70' }))
+		else if ([ 'chrome', 'chromium' ].includes(browserName) && browserVersion >= 70)
 		{
 			return 'Chrome70';
 		}
-		else if (browser.satisfies({ chrome: '>=67', chromium: '>=67' }))
+		else if ([ 'chrome', 'chromium' ].includes(browserName) && browserVersion >= 67)
 		{
 			return 'Chrome67';
 		}
-		else if (browser.satisfies({ chrome: '>=55', chromium: '>=55' }))
+		else if ([ 'chrome', 'chromium' ].includes(browserName) && browserVersion >= 55)
 		{
 			return 'Chrome55';
 		}
 		// Firefox.
-		else if (browser.satisfies({ firefox: '>=60' }))
+		else if (browserName === 'firefox' && osName !== 'ios' && browserVersion >= 60)
 		{
 			return 'Firefox60';
 		}
-		// Firefox on iOS.
-		else if (browser.satisfies({ ios: { OS: '>=14.3', firefox: '>=30.0' } }))
+		// Firefox on iOS (so Safari).
+		else if (browserName === 'firefox' && osName === 'ios' && osVersion >= 14.3)
```

Comment:
Is the firefox version check `firefox: '>=30.0'` drop intended

Reply:
I don't remember what such a "Firefox version 30" was honestly. Are you requesting this change based on previous code? Or because you know it makes sense? The thing is: if we are checking that iOS version is >= 14.3 then why do we need to also  check Firefox version? We know it's Safari anyway and we know is a modern version of it.

Reply:
I'm gonna merge and release. If you still think this is a required change, please comment later.

Reply:
Yeah I just wanted to make sure if the drop was intentional, I don't see a reason for including a firefox version as well